### ISSUE TITLE
Cannot exactly match custom attribute in search query

### ISSUE THREAD

**Issue by facetoe** [alerta/alerta#1286]
**Issue Summary**

Cannot exactly match a custom attribute value using the search syntax. As an example, the following query:

```
q=attributes.partition:7
```

The documentation seems to indicate this would be an exact match - https://docs.alerta.io/en/latest/api/query-syntax.html#nested-field-names.

Matches (7, 27, 17) etc. 

From the errors it is throwing when I pass invalid data, it looks like it is pattern matching in postgres with `~*`.

**Environment**
- OS: Linux
- API version: v8.0.2-0-gc2a4b9dea66d85364f2ddfdb93ee806af33da303
- Deployment: self-hosted
- For self-hosted, WSGI environment: gunicorn

- Database: Postgres

- Server config:
  Auth enabled? Yes
  Auth provider? Basic
  Customer views? No

**To Reproduce**
1. create some alerts with a number under a custom attribute
2. attempt to query only those with that exact number

**Additional context**
I may well be doing it wrong, but I have a query like:

```
q=(attributes.partition:7+AND+event:"my_event")+AND+(status:open)
```
basically want (partition=7 AND event=my_event AND status=open).

Any help would be greatly appreciated!

**Comment by satterly** [comment]
Try surrounding the 7 in double quotes. See https://docs.alerta.io/en/latest/api/query-syntax.html#id3

**Comment by facetoe** [comment]
Unfortunately this doesn't work and results in the following exception: 

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.7/site-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/lib/python3.7/site-packages/alerta/auth/decorators.py", line 53, in wrapped
    return f(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/alerta/models/metrics.py", line 258, in wrapped
    response = f(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/alerta/utils/response.py", line 18, in decorated
    return func(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/alerta/views/alerts.py", line 299, in search_alerts
    severity_count = Alert.get_counts_by_severity(query)
  File "/usr/lib/python3.7/site-packages/alerta/models/alert.py", line 491, in get_counts_by_severity
    return db.get_counts_by_severity(query)
  File "/usr/lib/python3.7/site-packages/alerta/database/backends/postgres/base.py", line 502, in get_counts_by_severity
    return {s.severity: s.count for s in self._fetchall(select, query.vars)}
  File "/usr/lib/python3.7/site-packages/alerta/database/backends/postgres/base.py", line 1413, in _fetchall
    cursor.execute(query, vars)
  File "/usr/lib64/python3.7/site-packages/psycopg2/extras.py", line 313, in execute
    return super(NamedTupleCursor, self).execute(query, vars)
psycopg2.ProgrammingError: operator does not exist: jsonb ~* unknown
LINE 3:              WHERE ("attributes" ~* '\y7\y' AND "event" ~* '...
                                         ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
```

I think it ends up hitting this point in the grammar as it is a custom attribute - https://github.com/alerta/alerta/blob/master/alerta/database/backends/postgres/queryparser.py#L69.

**Comment by satterly** [comment]
I think surrounding in quotes is the correct syntax and that this is just missing from my implementation. 

An alternative is to use explicit delimiters ...

> Define '^' as the first character of a search phrase meaning "Starts With".
Define '$' as the last character of a search phrase meaning "Ends With".
Pre-parse your search phrase, substituting '^' and '$' appropriately.

http://blogs.perl.org/users/mark_leighton_fisher/2012/01/stupid-lucene-tricks-exact-match-starts-with-ends-with.html

Either way, it needs work to fix. Thanks for reporting the issue. I'll add it to the backlog.

**Comment by satterly** [comment]
Surrounding with quotes will ensure an exact match now. For example ...
```
/alerts?q=_.partition:"7"
```
